pactman/verifier/verify.py

```diff
@@ -90,14 +107,30 @@ def service_PATCH(self, service_url):
         r = requests.patch(self._get_url(service_url), json=self._request_payload, headers=self._request_headers)
         return self.response.verify(r)
 
-    def setup(self, setup_url):
+    def set_provider_state(self, provider_setup):
+        if self.providerState is not None:
+            log.debug(f'Setting up provider state {self.providerState!r}')
+            return provider_setup(self.providerState)
+        for state in self.providerStates:
+            if 'name' not in state:
```

Comment:
```
try:
    name = state['name']
except KeyError:
    raise KeyError(f'provider state missing "name": {state}')
```
EAFP over LBYL ?
Should it be KeyError instead of ValueError ?

Reply:
The intent here is to provide an exception with more information than just a bare KeyError, given it's probably a common error. But yes, ValueError.